### PR TITLE
Feature/populate internal

### PR DESCRIPTION
**Pull request by ygworldr**
Internal server population changes.

### CODE REVIEW

File: cmd/mattermost/commands/populate_sample.go [hunk 1]
```diff
@@ -701,6 +774,50 @@ func createUserW(
...
+
+	team := createTeamMembershipW(channels, &teamName, teamRoles, channelRoles)
+	team.Categories = categories
+	teams := []app.UserTeamImportData{team}
+
+	user := app.UserImportData{
+		ProfileImage:       nil,
+		Username:           &usr.Username,
+		Email:              &usr.Email,
+		Password:           &usr.Username, // ☠ ☠ ☠  This is BAD, SO BAD!!!
```
Comment (Aurora12) [inline cmd/mattermost/commands/populate_sample.go:796]: Let's just replace it with something less obvious. Anyway, the users are supposed to change their passwords after logging in. Btw, that has to be a feature! Force users to change their passwords!

**Comment by ygworldr** [comment]
_**This, right there is what happens when you are in a hurry and use `--no-validate` to commit!!!**_ ☠☠☠☠☠☠☠☠☠☠☠☠☠☠…

File: cmd/mattermost/commands/populate_sample.go [hunk 2]
```diff
@@ -133,7 +174,8 @@ func populateSampleCmdF(command *cobra.Command, args []string) error {
 	defer a.Shutdown()
 
 	admins, err := command.Flags().GetStringSlice(PARAM_ADMINS)
-	if err != nil || len(admins) == 0 {
+	// if err != nil || len(admins) == 0 {
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (ygworldr) [reply cmd/mattermost/commands/populate_sample.go:178]: The `admins` option is not necessary. Those could (should) come from the configuration file.

File: cmd/mattermost/commands/populate_sample.go [hunk 3]
```diff
@@ -187,6 +229,64 @@ func populateSampleCmdF(command *cobra.Command, args []string) error {
...
 
+	// Get data from a configuration file.
+	// TODO: this should define ALL the options.
+	var initData InitialData
+	configurationFilePath, err := command.Flags().GetString(PARAM_CONFIG_FILE)
+	if fileExists(configurationFilePath) {
+		fmt.Println("Example file exists")
+		cfgData, err := ioutil.ReadFile(configurationFilePath)
+		if err != nil {
+			fmt.Print(err)
```
[the remark this answers is not in the document]
Comment (ygworldr) [reply cmd/mattermost/commands/populate_sample.go]: That is fair. I will add this.

File: cmd/mattermost/commands/populate_sample.go [hunk 4]
```diff
@@ -187,6 +229,64 @@ func populateSampleCmdF(command *cobra.Command, args []string) error {
 		}
 	}
 
+	// Get data from a configuration file.
+	// TODO: this should define ALL the options.
+	var initData InitialData
+	configurationFilePath, err := command.Flags().GetString(PARAM_CONFIG_FILE)
+	if fileExists(configurationFilePath) {
```
[the remark this answers is not in the document]
Comment (ygworldr) [reply cmd/mattermost/commands/populate_sample.go:236]: Yes, that is fair. Check added.

File: cmd/mattermost/commands/populate_sample.go [hunk 5]
```diff
@@ -187,6 +229,64 @@ func populateSampleCmdF(command *cobra.Command, args []string) error {
...
+		fmt.Println("Example file exists")
+		cfgData, err := ioutil.ReadFile(configurationFilePath)
+		if err != nil {
+			fmt.Print(err)
+		}
+		err = json.Unmarshal(cfgData, &initData)
+		if err != nil {
+			fmt.Println("error:", err)
+		}
+	} else {
```
[the remark this answers is not in the document]
Comment (ygworldr) [reply cmd/mattermost/commands/populate_sample.go:246]: Yes, added.

File: cmd/mattermost/commands/populate_sample.go (hunk 1 above)
Comment (ygworldr) [reply cmd/mattermost/commands/populate_sample.go:796]: Well, it is what the code does now. Plus, clearly, we will need a feature to allow users to change their password so I guess that can stay for now. I mean, we are not even going to do that for the FFF testers.

I *could* add that as a JSON option but then the password is in the clear there.

No good choices til there is a "reset" feature implemented.

Comment (ygworldr) [reply cmd/mattermost/commands/populate_sample.go:796]: @Aurora12 quoth on chat:
> Yeah, alright. We can do something smart later, like generate random passwords and output the list of usernames/passwords somewhere where the administrator can take and distribute them among users.

File: cmd/mattermost/commands/populate_sample.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ygworldr) [reply cmd/mattermost/commands/populate_sample.go]: I moved the option and added an extra check.

File: cmd/mattermost/commands/populate_sample.go [hunk 6]
```diff
@@ -187,6 +219,77 @@ func populateSampleCmdF(command *cobra.Command, args []string) error {
 		}
 	}
 
+	// Get data from a configuration file.
+	// TODO: this should define ALL the options.
+	var initData InitialData
+	configurationFilePath, err := command.Flags().GetString(PARAM_CONFIG_FILE)
+	if configurationFilePath != "" {
+		if fileExists(configurationFilePath) {
```
Comment (Aurora12) [inline cmd/mattermost/commands/populate_sample.go:227]: Else fail immediately! The user wanted to pass a configuration file and it wasn't found, we can't continue!

Comment (ygworldr) [reply cmd/mattermost/commands/populate_sample.go:227]: Fixed.

**Pull request merged**